iothub/service/API design.md

```diff
@@ -0,0 +1,87 @@
+```csharp
+
+public class IotHubServiceClient : IDisposable
+{
+	public DevicesClient Devices { get; private set; }
+	public ModulesClient Modules { get; private set; }
+
+	protected IotHubServiceClient();
+	
+	public IotHubServiceClient(string connectionString, IotHubServiceClientOptions options = default);
+	public IotHubServiceClient(string hostName, TokenCredential credential, IotHubServiceClientOptions options = default);
+	public IotHubServiceClient(string hostName, AzureSasCredential credential, IotHubServiceClientOptions options = default);
+
+	public void Dispose();
+}
+
+public class IotHubServiceClientOptions
+{
+	public IotHubServiceClientOptions(ServiceVersion version = LatestVersion)
+
+	public IWebProxy Proxy { get; set; }
+	public HttpClient HttpClient { get; set; }
+	public ServiceVersion Version { get; set; } = LatestVersion;
+
+	public enum ServiceVersion
```

Comment:
Think we should really do this?

Reply:
Track 2 guidance suggests we should. I can't say we've had a customer ever ask for it though

Reply:
Fair but:

1. We're not doing a full track2 client.
2. Even Krzyszstof/Jeff tell me they aren't sure of that recommendation and have considered removing it.

I just doubt the worth the hassle/complexity for this v2 client.

Reply:
Then I'll remove it for now and we can add it later if needed